### DISCUSSION TITLE
How to play background music while playing games?

### DISCUSSION THREAD

**Discussion by Pakman450** [OnionUI/Onion#1132] [Q&A]
Hey guys, I have Onion OS 4.2.0 on my MM+. I got really excited about the music player. However, I thought I could play music in the background while playing games. Before I suggest a feature, Is there a way to play our own music while playing our games?

btw, great os guys. It is _**way**_ better than stock os.

**Comment by Rahhide** [comment]
i think need an upgrade hardware to possible play music in background, i hope miyoo release the  miyoo pro version with 2 gigs ram and a high cpu processor. or maybe mod the retro arch to possible play music while playing.

**Comment by Strgnoff** [comment]
Hi everyone, reviving this 2023 thread since it’s now 2025 and GMU has matured and music playback has been taken more seriously, including [button lock-out](https://github.com/OnionUI/Onion/discussions/1775) as of 2025.

I've tested GMU on the Miyoo Mini Plus - playback works fine until you switch to another task, then it stops. This made me wonder whether shared access to the audio hardware (e.g. music + game audio) might be possible in principle.

The OS is already capable of running multiple background daemons (HTTP, FTP, Telnet, SSH, VNC) concurrently with games, so some multitasking clearly exists. I tested the VNC server together with gaming - it worked flawlessly.

The acute question now is whether the audio driver or kernel allows concurrent use of the audio device, or if it’s locked per process.

I understand the hardware limits (128 MB RAM), but a barebones MP3 service with no GUI or DSP may only very little RAM. For example, [walkgeek](https://github.com/oleg-it-com/walkgeek) is designed for embedded ARM platforms with only 192 KB SRAM. Maybe it could be compiled for OnionOS and Miyoo Mini Plus?

Would be great if any OnionOS developers or hardware experts could share insights on whether shared audio access is technically possible.

Example use-case: disable announcer and in-game music and listen to MP3 podcasts in the background with less story-driven games, e.g. with racing games like Colin McRae Rally 2.0 (via PS1 emulation).
